src/adaptor/mod.rs

```diff
@@ -0,0 +1,21 @@
+mod ini;
+
+use std::collections::HashMap;
+use std::io::{Read, Write};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum Config {
```

Comment:
In general, I try to avoid using a type from a library to represent my data model. It makes things easier down the road if we need to change the model. It also prevents the library from changing things on us. Plus, with Rust, it's very easy to write a simple converter function.